bson/src/main/org/bson/codecs/pojo/annotations/BsonId.java

```diff
@@ -32,6 +32,6 @@
  */
 @Documented
 @Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER})
+@Target({ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER, ElementType.RECORD_COMPONENT})
```

Comment:
Just confirming that after review of the Java 15 changes to the record specification, that this change is still necessary.  My reading suggests that we can do without it.

Reply:
This change is actually still necessary. The cleanest way to get the annotations for me is to get them off the RecordCompnent, because that's what I iterate over in PojoBuilderHelper::configureRecordClassModelBuilder. I've manually tested and confirmed that I cannot get them off the RecordComponent in Java 15 without targeting the RECORD_COMPONENT.

I believe what Java 15 allows is for you to get the annotation off the field (or method, or parameter, etc.) if it is applicable. For example, if you have a record like the following:
`
public record Holder(@BsonIgnore String field) {}
`
You could get the annotation by doing: `Holder.class.getDeclaredField("field").getAnnotations()` instead of having to go through the RecordComponent. 

It is possible for us to avoid targeting RECORD_COMPONENT if that is desirable (I would just have to lookup the field by its name and get the annotation from there), but the code will be a little less clean. Let me know what you think is best...

Reply:
I see now.  Thanks for the explanation.  I think it's fine to leave it as is.